pandas/core/indexing.py

```diff
@@ -1879,15 +1889,17 @@ def _has_valid_type(self, key, axis):
                                               "is not available")
                 raise ValueError("iLocation based boolean indexing cannot use "
                                  "an indexable as a mask")
-            return True
+            return
 
         if isinstance(key, slice):
-            return True
+            return
         elif is_integer(key):
-            return self._is_valid_integer(key, axis)
+            assert(self._is_valid_integer(key, axis))
```

Comment:
what is the purpose of this assert?

Reply:
Basically, ``_validate_key`` should raise an error if the keys are not valid (and they aren't if they are an integer but not a valid integer). Better to test and eventually raise a ``ValueError``?

Reply:
yes this should go thru ``_validate_key`` (the old ``_has_valid_type``). it should raise if indicated by that type of indexer, so yes test for that